packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -213,30 +224,102 @@ handler.checkIfPRIsInvalid = async function checkIfPRIsInvalid(
   }
 };
 
+/**
+ * Checks if a branch with a PR has branch protection, if not, comments on PR
+ * @param owner type string
+ * @param repo type string
+ * @param prNumber type number
+ * @param github type githup API surface from payload
+ */
+handler.checkForBranchProtection = async function checkForBranchProtection(
+  owner: string,
+  repo: string,
+  prNumber: number,
+  github: InstanceType<typeof ProbotOctokit>
+): Promise<string[] | undefined> {
+  let branchProtection: string[] | undefined;
+  // Check to see if branch protection exists
+  try {
+    branchProtection = (
+      await github.repos.getBranchProtection({
+        owner,
+        repo,
+        branch: 'master',
+      })
+    ).data.required_status_checks.contexts;
+    logger.info(
+      `checking branch protection for ${owner}/${repo}: ${branchProtection}`
+    );
+    // if branch protection doesn't exist, leave a comment on the PR;
+  } catch (err) {
+    err.message = `Error in getting branch protection\n\n${err.message}`;
+    await github.issues.createComment({
+      owner,
+      repo,
+      issue_number: prNumber,
+      body:
+        "Your PR doesn't have any required checks. Please add required checks to your master branch and then re-add the label. Learn more about enabling these checks here: https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks.",
+    });
+    logger.error(err);
+  }
+  return branchProtection;
+};
+
 /**
  * Adds a PR to datastore when an automerge label is added to a PR
  * @param url type string
  * @param wp type Watch PR (owner, repo, pr number, state, url)
  * @returns void
  */
-handler.addPR = async function addPR(wp: WatchPR, url: string) {
-  const key = datastore.key([TABLE, url]);
-  const entity = {
-    key,
-    data: {
-      created: new Date().toJSON(),
-      owner: wp.owner,
-      repo: wp.repo,
-      number: wp.number,
-      branchProtection: wp.branchProtection,
-      label: wp.label,
-      author: wp.author,
-      reactionId: wp.reactionId,
-      installationId: wp.installationId,
-    },
-    method: 'upsert',
-  };
-  await datastore.save(entity);
+handler.addPR = async function addPR(
+  incomingPR: IncomingPR,
+  url: string,
+  github: InstanceType<typeof ProbotOctokit>
+) {
+  let branchProtection: string[] | undefined;
+  try {
+    branchProtection = await handler.checkForBranchProtection(
```

Comment:
It feels a little strange to me that we're checking `branchProtection` before adding the PR, since this is something we check in `merge-logic.ts` right?

From a separation of functionality point of view, I'd be tempted to try to keep all the logic for if something _should_ be merged in `merge-logic.ts`, whereas `merge-on-green.ts` just handles:

* receiving Webhooks from GitHub 
* adding PRs to the DB (either based on webhook or your periodic cron job).

Reply:
Yeah! I think that was the thinking behind our original logic. The reason I changed where we check for branch protection is because it's a way of reducing API requests during the main` merge-logic`, especially since branch protection isn't really the kind of information that changes within the life of a PR (so it's not the kind of thing we need to check multiple times before merging a PR). The way we currently have it set up allows us to only check for branch protection once, and then we save it in our Datastore table and don't have to make that API call again. Another reason (though this one is weaker) is that it kind of makes sense for us to only add a PR to our queue of PRs to check if it has branch protection. In other words, instead of: (a) adding a PR to datastore, (b) checking if it has branch protection, (c) deleting the PR, we could save ourselves two steps and just never add the PR to Datastore if it doesn't have branch protection. Open to hearing your thoughts on this process!

Reply:
My temptation would be to still have the logic for checking branch protection inside of the `merge-logic.js` class, and you could just call that method here (rather than having the logic in two places).

This doesn't need to be a blocker for this PR, but perhaps we could make a tracking ticket to do this?